### PR TITLE
Add File entity

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -2632,6 +2632,21 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
...
+            'name': entity_fields.StringField(unique=True),
+            'path': entity_fields.StringField(),
+            'uuid': entity_fields.StringField(),
+            'checksum': entity_fields.StringField(),
+            'repository': entity_fields.OneToOneField(Repository),
+        }
+        self._meta = {'api_path': 'katello/api/v2/files'}
+        super(File, self).__init__(server_config, **kwargs)
+
+
```
Comment (omkarkhatavkar) [inline nailgun/entities.py]: Tested, works well 

```
In [13]: files = entities.File().search(query={'per_page': 1000})

In [14]: files
Out[14]: 
[nailgun.entities.File(id=1, checksum='b84a8551f408c46ad0000a99275dff796af71b752a14dc56c91f56e22c88165d', path='hammer_debug_log.txt', name='hammer_debug_log.txt', uuid='58c44b8a-599d-4c2d-b8cd-0e2ee3b6314b'),
 nailgun.entities.File(id=2, checksum='6dad84effda2de681476a26613299bcb99bcd7b6be3f4a58d0e9a3df25cbd77e', path='Screenshot from 2018-10-30 12-12-14.png', name='Screenshot from 2018-10-30 12-12-14.png', uuid='ec7a0f27-670a-4a9e-8782-0483f2add27f')]

In [15]: files[0].name
Out[15]: 'hammer_debug_log.txt'

In [16]: len(files)
Out[16]: 2
```

Comment (omkarkhatavkar) [reply nailgun/entities.py]: 1. Maybe we can also update the missing file entity at same as we did for packages. puppet modules here https://github.com/SatelliteQE/nailgun/blob/0de6643ca1d4048aaca82d03e50dd07dd3faa58b/nailgun/entities.py#L5408

Comment (sean797) [reply nailgun/entities.py]: @omkarkhatavkar added - thanks :+1:

**Comment by omkarkhatavkar** [comment]
@sean797 :+1:  only coverage is decreased if that fixed, I'm good to ACK

**Comment by san7ket** [comment]
@sean797 Can we also have a short test for file filters please. Looks like that's the reason the coverage decreased.

**Comment by sean797** [comment]
Updated & @coveralls seems happy, thanks :+1:

**Pull request merged**